API/SSW.Rewards.Application/Reward/Queries/GetRecentRewards/GetRecentRewardsQueryHandler.cs

```diff
@@ -0,0 +1,54 @@
+﻿using AutoMapper;
+using AutoMapper.QueryableExtensions;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using SSW.Rewards.Application.Common.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SSW.Rewards.Application.Reward.Queries.GetRecentRewards
+{
+    public class GetRecentRewardsQueryHandler : IRequestHandler<GetRecentRewardsQuery, RecentRewardListViewModel>
+    {
+        private IMapper _mapper;
+        private ISSWRewardsDbContext _context;
+
+        public GetRecentRewardsQueryHandler(
+            IMapper mapper,
+            ISSWRewardsDbContext context)
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+
+        public async Task<RecentRewardListViewModel> Handle(GetRecentRewardsQuery request, CancellationToken cancellationToken)
+        {
+            var query = _context.UserRewards.Where(u => u.Id != 0);
+
+            if(!(request.Since == null))
```

Comment:
You can simplify this logic and swap the two blocks of code... avoids the !

Reply:
Thanks Will - would appreciate your guidance here. I did originally have a simpler version of this, but the bit that threw me off was the need to have the `Take(10)` included in one condition but not the other.

Is your recommendation to streamline this into a single code block? Because that's where I struggled. Or are you suggesting avoid the `if(!(request.Since == null))`? So it would change to:

```
if(request.Since == null)
{
         query = query
             .OrderByDescending(u => u.AwardedAt)
             .Take(10);
}
else
{
         query = query
            .Where(u => u.AwardedAt > request.Since.Value.ToUniversalTime());
}
```